packages/contractkit/src/wrappers/Attestations.ts

```diff
@@ -583,6 +575,19 @@ export class AttestationsWrapper extends BaseWrapper<Attestations> {
     })
   }
 
+  /**
+   * Returns attestation code for provided security code from validator's attestation service
+   * @param serviceURL: validator's attestation service URL
+   * @param body: onServiceCodeForSecurityRequest
+   */
+  getAttestationForSecurityCode(serviceURL: string, body: AttesationServiceCodeForSecurityRequest) {
+    const urlParams = new URLSearchParams({ ...body })
+    return fetch(appendPath(serviceURL, 'get_attestations') + '?' + urlParams, {
```

Comment:
Does this include any retry logic? We should add this (with exponential backoff) if not.

Reply:
Because we have potentially multiple requests for the same security code and some requests we expect to fail I'm not sure that this is a good idea. WDYT?

Reply:
I'm having a hard time to understand when do we want to do that. Basically the fact that we are not 100% sure that current validator is the one who sent the code brings some complexity here. Possible cases I see:

* Timeout - is it really worth trying it once again considering that it might not be the one we need.
* `4xx` - we expect this to happen for some validators.
* `5xx` - something went really wrong on the attestation service side. Probably, it make sense to try it later, but how much time we want to spend on each validator in total?

Any thoughts? cc: @aslawson @timmoreton @codyborn

Reply:
Perhaps we can parallelize the requests.. if we get a 200 response from one of them we return immediately. Otherwise, for each 500 or timeout we retry a couple times before giving up.

Reply:
This would be an optimization. We do have manual retry by the user.  If there is a failure and user selects "resend", it will try the existing actionable issuer first again. We allow the user to do this after 1 min, but if we receive either a timeout or a failure (other than expected failure on "already sent too many times" or "wrong code") we could resend the request within that minute.

Reply:
actually i suppose that's for the initial request for get the security code, so really only "wrong code" would be the acceptable error

Reply:
I think it would be nice if we can retry with backoff except on cases where we get 4xx -- especially 403 (security code invalid, or attempts exceeded) or 404 (attestation not found -- i.e it was a hash collision and this is the wrong service).  However I feel it is an optimization if time is tight